### PR TITLE
[BE][Easy] apply autofix for ruff rules unnecessary-collection-call (C408): `list()` / `tuple()` / `dict()`

### PR DESCRIPTION
**Pull request by XuehaiPan**
This PR changes the empty collection factory call to Python literals:

- `list()` -> `[]`
- `tuple()` -> `()`
- `dict()` -> `{}`

The Python literals are more performant and safer. For example, the bytecode for building an empty dictionary:

```bash
$ python3 -m dis - <<EOS
import collections

d1 = {}
d2 = dict()

dict = collections.OrderedDict
d3 = dict()
EOS
```

```text
  0           0 RESUME                   0

  1           2 LOAD_CONST               0 (0)
              4 LOAD_CONST               1 (None)
              6 IMPORT_NAME              0 (collections)
              8 STORE_NAME               0 (collections)

  3          10 BUILD_MAP                0
             12 STORE_NAME               1 (d1)

  4          14 PUSH_NULL
             16 LOAD_NAME                2 (dict)
             18 CALL                     0
             26 STORE_NAME               3 (d2)

  6          28 LOAD_NAME                0 (collections)
             30 LOAD_ATTR                8 (OrderedDict)
             50 STORE_NAME               2 (dict)

  7          52 PUSH_NULL
             54 LOAD_NAME                2 (dict)
             56 CALL                     0
             64 STORE_NAME               5 (d3)
             66 RETURN_CONST             1 (None)
```

The dict literal `{}` only has one bytecode `BUILD_MAP`, while the factory call `dict()` has three `PUSH_NULL + LOAD_NAME + CALL`. Also, the factory call is not safe if users override the `dict` name in `locals` or `globals` (see the example of replacing with `OrderedDict` above).

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130199

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @LucasLLC @MeetVadakkanchery @mhorowitz @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @tianyu-l

### CODE REVIEW

**Review by malfet — COMMENTED** [review]
Please provide some sort of argument why one is better than another. And also, with all those linter changes, I would like to see document of sorts, because I think "applying all linter rules under the sun" is a bad policy

**Comment by XuehaiPan** [comment]
> Please provide some sort of argument why one is better than another.

I updated the PR comment.

File: pyproject.toml [hunk 1]
```diff
@@ -76,7 +76,7 @@ ignore = [
     "SIM102", "SIM103", "SIM112", # flake8-simplify code styles
     "SIM113", # please fix
     "SIM105", # these ignores are from flake8-simplify. please fix or ignore with commented reason
-    "SIM108",
+    "SIM108", # SIM108 ignored because we prefer if-else-block instead of ternary expression
```
Comment (malfet) [inline pyproject.toml:79]: This seems unrelated to the changes in question, isn't it?

File: torch/_inductor/runtime/hints.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def instance_descriptor(
     instance_descriptor = collections.namedtuple(  # type: ignore[no-redef]
         "instance_descriptor",
         ["divisible_by_16", "equal_to_1", "ids_of_folded_args", "divisible_by_8"],
-        defaults=[tuple(), tuple(), tuple(), tuple()],
+        defaults=[(), (), (), ()],
```
Comment (malfet) [inline torch/_inductor/runtime/hints.py:71]: Not sure which one is faster, more readable to be frank
```suggestion
        defaults=[()] * 4,
```

File: torch/testing/_internal/opinfo/utils.py [hunk 3]
```diff
@@ -103,7 +103,7 @@ def dtypes_dispatch_hint(dtypes):
 
     # CUDA is not available, dtypes will be empty.
     if len(dtypes) == 0:
-        return return_type((), str(tuple()))
+        return return_type((), str(()))
```
Comment (malfet) [inline torch/testing/_internal/opinfo/utils.py:106]: `str(tuple())` is always `"()"`, isn't it?
```suggestion
        return return_type((), "()")
```

**Review by malfet — APPROVED** [review]
LGTM(and thank you for the detailed description), but please update `pyproject.toml` to include C408

**Comment by XuehaiPan** [comment]
> but please update `pyproject.toml` to include C408

`C408` is ignored because we prefer `dict(kw=arg)` syntax, e.g.:

```python
call(func, args=(a, b, c), kwargs=dict(arg1=x, arg2=y, arg3=z))
```

instead of

```python
call(func, args=(a, b, c), kwargs={"arg1": x, "arg2": y, "arg3": z})
```

https://github.com/pytorch/pytorch/blob/5db9bd467e692ad857bb2e678904352039385e6a/pyproject.toml#L55
